framework/src/mesh/StitchedMesh.C

```diff
@@ -0,0 +1,108 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "StitchedMesh.h"
+#include "Parser.h"
+#include "InputParameters.h"
+
+// libMesh includes
+#include "libmesh/mesh_modification.h"
+#include "libmesh/serial_mesh.h"
+#include "libmesh/exodusII_io.h"
+
+template<>
+InputParameters validParams<StitchedMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredParam<std::vector<MeshFileName>>("files", "The name of the mesh files to read.  These mesh files will be 'stitched' into the current mesh in this order.");
+
+  params.addRequiredParam<std::vector<BoundaryName>>("stitch_boundaries", "Pairs of boundary names (one after the other) to stitch together for each step.");
```

Comment:
> Is there an automatic way of identifying the stitch boundaries?

Not currently.  The libMesh functionality requires the boundary IDs.  Someone could certainly enhance the `ReplicatedMesh::stitch_meshes()` function in libMesh to do a search... but that's not going to happen anytime soon (that would also be incredibly slow... but if you needed it, I agree that it would be good to have).